database/seed.js

```diff
@@ -1,80 +1,99 @@
 
 // seed main product table
 
 let seedMainProduct = function () {
-    return new Promise (function (resolve, reject) {
+
+        var promisearr = [];
 
         for (var i = 0; i < 100; i++) {
-            
-            function Product(category, company, description, price, prime, rating, title, bestseller) {
-                this.category = randomCategory;
-                this.company = randomCompany;
-                this.description = randomDescription;
-                this.price = randomPrice;
-                this.prime = randomPrime;
-                this.rating = randomRating;
-                this.title = randomTitle;
-                this.bestseller = randomBestSeller;
-            }
-
-            const products = new Product(randomCategory, randomCompany, randomDescription, randomPrice, randomPrice, randomRating, randomTitle, randomBestSeller)
-
-            let unit = new mainProduct({
-                category: `${products.randomCategory}`,
-                company: `${products.randomCompany}`,
-                description: `${products.randomDescription}`,
-                price: `${products.randomPrice}`,
-                prime: `${products.randomPrime}`,
-                ratings: `${products.randomRating}`,
-                title: `${products.randomTitle}`,
-                best_seller: `${products.randomBestSeller}`
-            }).save(function (err) {
-                if(err) {
-                    console.log(err, 'err from .save')
-                    reject
-                } else {
-                    resolve();
-                }
-            })
-        }
 
-    });
+            // const products = new Product(randomCategory, randomCompany, randomDescription, randomPrice, randomPrice, randomRating, randomTitle, randomBestSeller)
+
+            let randomCategory = faker.commerce.department(); // Clothing
+            let randomCompany = faker.company.companyName(); // Apple Inc.
+            let randomDescription = faker.image.abstract(); // image description
+            let randomPrice = faker.commerce.price(); // $1.00
+            let randomPrime = faker.random.boolean(); // true
+            let randomRating = faker.random.number(); // 4
+            let randomTitle = faker.commerce.productName(); // Swimming Pool
+            let randomBestSeller = faker.random.boolean(); // false
+
+            let mainproduct = new mainProduct({
+                category: `${randomCategory}`,
+                company: `${randomCompany}`,
+                description: `${randomDescription}`,
+                price: `${randomPrice}`,
+                prime: `${randomPrime}`,
+                ratings: `${randomRating}`,
+                title: `${randomTitle}`,
+                best_seller: `${randomBestSeller}`
+            }).save();
+            promisearr.push(mainproduct)
+        }
+        return Promise.all(promisearr).catch(err => console.log(err))
```

Comment:
I like that you handle errors first with .catch so that even if one of the steps goes wrong when you invoke the function later on the database connection will still always get closed.